pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -280,11 +282,41 @@ type ParallelSteps struct {
 	Steps []WorkflowStep `protobuf:"bytes,1,rep,name=steps"`
 }
 
+// WorkflowStep is an anonymous list inside of ParallelSteps (i.e. it does not have a key), so it needs its own
+// custom Unmarshaller
 func (p *ParallelSteps) UnmarshalJSON(value []byte) error {
-	err := json.Unmarshal(value, &p.Steps)
+	// Since we are writing a custom unmarshaller, we have to enforce the "DisallowUnknownFields" requirement manually.
+
+	// First, get a generic representation of the contents
+	var candidate []map[string]interface{}
+	err := json.Unmarshal(value, &candidate)
```

Comment:
unit test?

Reply:
This is tested directly when workflows are unmarshalled. As an example from this PR: https://github.com/argoproj/argo/pull/2210/files#diff-fa83dc0ba3700366cb26786c8d3f381dR68